node/core/runtime-api/src/lib.rs

```diff
@@ -486,6 +492,8 @@ where
 		Request::PvfsRequirePrecheck(sender) => {
 			query!(PvfsRequirePrecheck, pvfs_require_precheck(), ver = 2, sender)
 		},
+		Request::ValidationCodeHash(para, assumption, sender) =>
+			query!(ValidationCodeHash, validation_code_hash(para, assumption), ver = 2, sender),
```

Comment:
I think it would be possible to implement a fallback here instead of on the caller side as done in #4628, but since it's used only in one place, no biggie.

Reply:
Yeah, I was on the fence about this either. I picked this one because theoretically, in the general case, the user knows better what to do, and I did not want to mix into runtime-api a more higher level logic. But I did not spend too much time thinking about it, since I expect this fallback soon to be not required at all and it's only one user now.

Reply:
Make sense :+1: